src/ORM/Association/HasMany.php

```diff
@@ -197,9 +197,10 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
      * target entity, and the parent entity.
      * @param \Cake\Datasource\EntityInterface $parentEntity The source entity containing the target
      * entities to be saved.
-     * @param array $entities list of entities to persist in target table and to
-     * link to the parent entity
+     * @param \Cake\Datasource\EntityInterface[]|\Cake\Datasource\InvalidPropertyInterface[] $entities list of entities
```

Comment:
It should be `(\Cake\Datasource\EntityInterface&\Cake\Datasource\InvalidPropertyInterface)[]`